src/main/java/com/dev/cinema/util/HashUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.dev.cinema.util;
+
+import com.dev.cinema.exceptions.PasswordHashingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import org.apache.log4j.Logger;
+
+public class HashUtil {
+    private static final Logger LOGGER = Logger.getLogger(HashUtil.class);
+    private static final String HASHING_ALGORITHM = "SHA-512";
+    private static final String MESSAGE =
+            "Password hashing failed. Unable to find the chosen algorithm: " + HASHING_ALGORITHM;
+
+    public static byte[] getSalt() {
+        SecureRandom secureRandom = new SecureRandom();
+        byte[] salt = new byte[16];
+        secureRandom.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(HASHING_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte element : digest) {
+                hashedPassword.append(String.format("%02x", element));
+            }
+        } catch (NoSuchAlgorithmException exception) {
+            LOGGER.error(exception);
+            throw new PasswordHashingException(MESSAGE, exception);
```

Comment:
there is no need to throw checked exception in this situation, you may use runtime

Reply:
My exception extends Runtime
May I leave it?

Reply:
in this case  - yes, but why is it then in login() method signature ?

Reply:
Memory issues :D